### ISSUE TITLE
LS49 and xfel_regression test failures

### ISSUE THREAD

**Issue by dwpaley** [cctbx/cctbx_project#712]
After commit 2243cc9, some XFEL tests are failing. The LS49 failures occur when simulating a diffraction image (using structure factors calculated from a model) and then comparing the result to a reference. The xfel_regression failures presumably also involve (1) doing something with computed structure factors and (2) comparing results to a reference. A list of failures is here:

<details>

```
2021-12-20T00:32:45.5858219Z Tests finished. Elapsed time: 3102.82s
2021-12-20T00:32:45.5858495Z 
2021-12-20T00:32:45.5863115Z Distribution of test runtimes:
2021-12-20T00:32:45.5864550Z   0.6s - 107.6s: 79
2021-12-20T00:32:45.5865076Z   107.6s - 214.7s: 8
2021-12-20T00:32:45.5869406Z   214.7s - 321.8s: 1
2021-12-20T00:32:45.5870105Z   321.8s - 428.8s: 2
2021-12-20T00:32:45.5870965Z   428.8s - 535.9s: 0
2021-12-20T00:32:45.5871514Z   535.9s - 643.0s: 0
2021-12-20T00:32:45.5875014Z   643.0s - 750.0s: 0
2021-12-20T00:32:45.5875454Z   750.0s - 857.1s: 0
2021-12-20T00:32:45.5875954Z   857.1s - 964.2s: 0
2021-12-20T00:32:45.5881075Z   964.2s - 1071.2s: 2
2021-12-20T00:32:45.5881234Z 
2021-12-20T00:32:45.5881333Z 
2021-12-20T00:32:45.5882000Z Warning: the following jobs took at least 180 seconds:
2021-12-20T00:32:45.5883185Z   libtbx.python "/__w/1/modules/cctbx_project/simtbx/diffBragg/tests/tst_diffBragg_eta_derivs.py" --curvatures: 246.8s
2021-12-20T00:32:45.5887814Z   libtbx.python "/__w/1/modules/xfel_regression/test/command_line/tst_cspad_cbf_metrology.py" 300: 402.8s
2021-12-20T00:32:45.5891376Z   libtbx.python "/__w/1/modules/xfel_regression/test/command_line/tst_cxi_merge.py" 7: 417.7s
2021-12-20T00:32:45.5893809Z   libtbx.python "/__w/1/modules/LS49/tests/tst_cpu_add_spots_double_precision.py" poly: 1067.8s
2021-12-20T00:32:45.5894667Z   libtbx.python "/__w/1/modules/LS49/tests/tst_polychromatic_image.py": 1071.2s
2021-12-20T00:32:45.5898290Z Please try to reduce overall runtime - consider splitting up these tests.
2021-12-20T00:32:45.5898829Z 
2021-12-20T00:32:45.5899138Z 
2021-12-20T00:32:45.5905361Z Error: the following jobs returned non-zero exit codes or suspicious stderr output:
2021-12-20T00:32:45.5905782Z 
2021-12-20T00:32:45.5908225Z libtbx.python "/__w/1/modules/xfel_regression/test/command_line/tst_cxi_merge.py" 1 [FAIL] 142.4s
2021-12-20T00:32:45.5908649Z   Time: 142.44
2021-12-20T00:32:45.5912559Z   Return code: 1
2021-12-20T00:32:45.5923787Z   OKs: 0
2021-12-20T00:32:45.5924184Z libtbx.python "/__w/1/modules/xfel_regression/test/command_line/tst_cxi_merge.py" 2 [FAIL] 150.5s
2021-12-20T00:32:45.5926568Z   Time: 150.45
2021-12-20T00:32:45.5926839Z   Return code: 1
2021-12-20T00:32:45.5927149Z   OKs: 0
2021-12-20T00:32:45.5927536Z libtbx.python "/__w/1/modules/xfel_regression/test/command_line/tst_xfel_merge.py" all_2 [FAIL] 124.1s
2021-12-20T00:32:45.5932877Z   Time: 124.12
2021-12-20T00:32:45.5933198Z   Return code: 1
2021-12-20T00:32:45.5933453Z   OKs: 0
2021-12-20T00:32:45.5933885Z libtbx.python "/__w/1/modules/xfel_regression/test/command_line/tst_xfel_merge.py" postrefine [FAIL] 10.4s
2021-12-20T00:32:45.5934274Z   Time: 10.37
2021-12-20T00:32:45.5934589Z   Return code: 1
2021-12-20T00:32:45.5934837Z   OKs: 0
2021-12-20T00:32:45.5940979Z libtbx.python "/__w/1/modules/xfel_regression/test/command_line/tst_cxi_merge.py" 7 [FAIL] 417.7s
2021-12-20T00:32:45.5941500Z   Time: 417.71
2021-12-20T00:32:45.5941765Z   Return code: 1
2021-12-20T00:32:45.5942069Z   OKs: 0
2021-12-20T00:32:45.5942411Z libtbx.python "/__w/1/modules/LS49/tests/tst_jh_add_spots.py" [FAIL] 54.2s
2021-12-20T00:32:45.5942813Z   Time: 54.16
2021-12-20T00:32:45.5943073Z   Return code: 1
2021-12-20T00:32:45.5947084Z   OKs: 0
2021-12-20T00:32:45.5947439Z libtbx.python "/__w/1/modules/LS49/tests/tst_monochromatic_image.py" [FAIL] 29.8s
2021-12-20T00:32:45.5947850Z   Time: 29.81
2021-12-20T00:32:45.5948162Z   Return code: 1
2021-12-20T00:32:45.5948415Z   OKs: 0
2021-12-20T00:32:45.5948834Z libtbx.python "/__w/1/modules/LS49/tests/tst_cpu_add_spots_double_precision.py" mono [FAIL] 28.9s
2021-12-20T00:32:45.5949206Z   Time: 28.91
2021-12-20T00:32:45.5952895Z   Return code: 1
2021-12-20T00:32:45.5953152Z   OKs: 0
2021-12-20T00:32:45.5953553Z libtbx.python "/__w/1/modules/LS49/tests/tst_polychromatic_image.py" [FAIL] 1071.2s
2021-12-20T00:32:45.5953910Z   Time: 1071.23
2021-12-20T00:32:45.5958417Z   Return code: 1
2021-12-20T00:32:45.5958940Z   OKs: 0
2021-12-20T00:32:45.5959314Z libtbx.python "/__w/1/modules/LS49/tests/tst_cpu_add_spots_double_precision.py" poly [FAIL] 1067.8s
2021-12-20T00:32:45.5962869Z   Time: 1067.78
2021-12-20T00:32:45.5963137Z   Return code: 1
2021-12-20T00:32:45.5963445Z   OKs: 0
2021-12-20T00:32:45.5963697Z   Standard error:
2021-12-20T00:32:45.5966815Z     Traceback (most recent call last):
2021-12-20T00:32:45.5967325Z       File "/__w/1/modules/xfel_regression/test/command_line/tst_cxi_merge.py", line 100, in <module>
2021-12-20T00:32:45.5969420Z         tester.run_all(test_number)
2021-12-20T00:32:45.5972612Z       File "/__w/1/modules/xfel_regression/test/command_line/tst_cxi_merge.py", line 76, in run_all
2021-12-20T00:32:45.5973078Z         self.test_merge_themolysin(False, "sdfac_auto", (42.2, 33.0))
2021-12-20T00:32:45.5973618Z       File "/__w/1/modules/xfel_regression/test/command_line/tst_cxi_merge.py", line 68, in test_merge_themolysin
2021-12-20T00:32:45.5982766Z         assert cc == cc_onehalfs[1], (cc, cc_onehalfs[1])
2021-12-20T00:32:45.5983154Z     AssertionError: (33.1, 33.0)
2021-12-20T00:32:45.5983516Z   Standard error:
2021-12-20T00:32:45.5983836Z     Traceback (most recent call last):
2021-12-20T00:32:46.4440393Z       File "/__w/1/modules/xfel_regression/test/command_line/tst_cxi_merge.py", line 100, in <module>
2021-12-20T00:32:46.4444716Z         tester.run_all(test_number)
2021-12-20T00:32:46.4445279Z       File "/__w/1/modules/xfel_regression/test/command_line/tst_cxi_merge.py", line 78, in run_all
2021-12-20T00:32:46.4445917Z         self.test_merge_themolysin(False, "errors_from_sample_residuals", (75.1, 62.4))
2021-12-20T00:32:46.4446546Z       File "/__w/1/modules/xfel_regression/test/command_line/tst_cxi_merge.py", line 56, in test_merge_themolysin
2021-12-20T00:32:46.4447076Z         assert cc == cc_onehalfs[0], (cc, cc_onehalfs[0])
2021-12-20T00:32:46.4451185Z     AssertionError: (75.2, 75.1)
2021-12-20T00:32:46.4451715Z   Standard error:
2021-12-20T00:32:46.4453378Z     /__w/1/modules/xfel_regression/test/command_line/tst_xfel_merge.py:410: DeprecationWarning: distro.linux_distribution() is deprecated. It should only be used as a compatibility shim with Python's platform.linux_distribution(). Please use distro.id(), distro.version() and distro.name() instead.
2021-12-20T00:32:46.4454413Z       linux_distribution = distro_checker.linux_distribution(full_distribution_name=False)
2021-12-20T00:32:46.4454842Z     Traceback (most recent call last):
2021-12-20T00:32:46.4457953Z       File "/__w/1/modules/xfel_regression/test/command_line/tst_xfel_merge.py", line 435, in <module>
2021-12-20T00:32:46.4458423Z         tester.run_test(test_id)
2021-12-20T00:32:46.4458832Z       File "/__w/1/modules/xfel_regression/test/command_line/tst_xfel_merge.py", line 77, in run_test
2021-12-20T00:32:46.4461672Z         executor.run()
2021-12-20T00:32:46.4462246Z       File "/__w/1/modules/xfel_regression/test/command_line/tst_xfel_merge.py", line 249, in run
2021-12-20T00:32:46.4462680Z         self.compare_results_with_reference()
2021-12-20T00:32:46.4465307Z       File "/__w/1/modules/xfel_regression/test/command_line/tst_xfel_merge.py", line 269, in compare_results_with_reference
2021-12-20T00:32:46.4465877Z         self.compare_reflection_data(self.output_merged_data_file_path, self.reference_merged_data_file_path)
2021-12-20T00:32:46.4466716Z       File "/__w/1/modules/xfel_regression/test/command_line/tst_xfel_merge.py", line 388, in compare_reflection_data
2021-12-20T00:32:46.4470392Z         "Test %s: %s vs %s - Intensities are different (cc: %.1f)"%(self.params["id"], file_1, file_2, 100*cc)
2021-12-20T00:32:46.4471788Z     AssertionError: Test all_2: /__w/1/tests/tmpuqmm0p63_tst_xfel_merge_114_all_2/out/test_all_2_all.mtz vs /__w/1/modules/xfel_regression/merging_test_data/reference_data/test_all_2_all_reference_C7.mtz - Intensities are different (cc: 98.8)
2021-12-20T00:32:46.4472762Z   Standard error:
2021-12-20T00:32:46.4477058Z     /__w/1/modules/xfel_regression/test/command_line/tst_xfel_merge.py:410: DeprecationWarning: distro.linux_distribution() is deprecated. It should only be used as a compatibility shim with Python's platform.linux_distribution(). Please use distro.id(), distro.version() and distro.name() instead.
2021-12-20T00:32:46.4478562Z       linux_distribution = distro_checker.linux_distribution(full_distribution_name=False)
2021-12-20T00:32:46.4479184Z     Traceback (most recent call last):
2021-12-20T00:32:46.4482270Z       File "/__w/1/modules/xfel_regression/test/command_line/tst_xfel_merge.py", line 435, in <module>
2021-12-20T00:32:46.4482849Z         tester.run_test(test_id)
2021-12-20T00:32:46.4485727Z       File "/__w/1/modules/xfel_regression/test/command_line/tst_xfel_merge.py", line 77, in run_test
2021-12-20T00:32:46.4486350Z         executor.run()
2021-12-20T00:32:46.4486802Z       File "/__w/1/modules/xfel_regression/test/command_line/tst_xfel_merge.py", line 249, in run
2021-12-20T00:32:46.4487288Z         self.compare_results_with_reference()
2021-12-20T00:32:46.4492693Z       File "/__w/1/modules/xfel_regression/test/command_line/tst_xfel_merge.py", line 282, in compare_results_with_reference
2021-12-20T00:32:46.4493345Z         self.compare_experiments(self.output_expt_file_path, self.reference_expt_file_path)
2021-12-20T00:32:46.4493920Z       File "/__w/1/modules/xfel_regression/test/command_line/tst_xfel_merge.py", line 307, in compare_experiments
2021-12-20T00:32:46.4503071Z         (file_1, file_2, len(expts_1), len(expts_2))
2021-12-20T00:32:46.4504078Z     AssertionError: Experiment files contain different number of experiments (/__w/1/tests/tmpdwbs51v2_tst_xfel_merge_114_postrefine/out/test_postrefine.expt vs /__w/1/modules/xfel_regression/merging_test_data/reference_data/test_postrefine_reference_C7.expt, 31 vs. 30)
2021-12-20T00:32:46.4505094Z   Standard error:
2021-12-20T00:32:46.4505545Z     Traceback (most recent call last):
2021-12-20T00:32:46.4506119Z       File "/__w/1/modules/xfel_regression/test/command_line/tst_cxi_merge.py", line 100, in <module>
2021-12-20T00:32:46.4511370Z         tester.run_all(test_number)
2021-12-20T00:32:46.4511806Z       File "/__w/1/modules/xfel_regression/test/command_line/tst_cxi_merge.py", line 88, in run_all
2021-12-20T00:32:46.4512327Z         self.test_merge_themolysin(False, "sdfac_refine", (56.5, 46.3))
2021-12-20T00:32:46.4512811Z       File "/__w/1/modules/xfel_regression/test/command_line/tst_cxi_merge.py", line 56, in test_merge_themolysin
2021-12-20T00:32:46.4513352Z         assert cc == cc_onehalfs[0], (cc, cc_onehalfs[0])
2021-12-20T00:32:46.4517761Z     AssertionError: (57.0, 56.5)
2021-12-20T00:32:46.4518115Z   Standard error:
2021-12-20T00:32:46.4518529Z     Traceback (most recent call last):
2021-12-20T00:32:46.4518936Z       File "/__w/1/modules/LS49/tests/tst_jh_add_spots.py", line 17, in <module>
2021-12-20T00:32:46.4519731Z         compare_two_images(reference=os.path.join(ls49_big_data,"reference","step5_000000.img.gz"), test="./jh_step5_000000.img.gz")
2021-12-20T00:32:46.4520537Z       File "/__w/1/modules/LS49/tests/tst_monochromatic_image.py", line 70, in compare_two_images
2021-12-20T00:32:46.4521125Z         assert N_large_deltas == 0, "%d pixels have |delta| larger than cutoff %d"%(N_large_deltas,tolerance_delta)
2021-12-20T00:32:46.4527523Z     AssertionError: 1862 pixels have |delta| larger than cutoff 50
2021-12-20T00:32:46.4528470Z   Standard error:
2021-12-20T00:32:46.4528873Z     Traceback (most recent call last):
2021-12-20T00:32:46.4529418Z       File "/__w/1/modules/LS49/tests/tst_monochromatic_image.py", line 105, in <module>
2021-12-20T00:32:46.4530223Z         compare_two_images(reference=os.path.join(ls49_big_data,"reference","step5_000000.img.gz"), test="./step5_000000.img.gz")
2021-12-20T00:32:46.4530833Z       File "/__w/1/modules/LS49/tests/tst_monochromatic_image.py", line 70, in compare_two_images
2021-12-20T00:32:46.4539811Z         assert N_large_deltas == 0, "%d pixels have |delta| larger than cutoff %d"%(N_large_deltas,tolerance_delta)
2021-12-20T00:32:46.4540704Z     AssertionError: 1862 pixels have |delta| larger than cutoff 50
2021-12-20T00:32:46.4541158Z   Standard error:
2021-12-20T00:32:46.4541638Z     Traceback (most recent call last):
2021-12-20T00:32:46.4542174Z       File "/__w/1/modules/LS49/tests/tst_cpu_add_spots_double_precision.py", line 49, in <module>
2021-12-20T00:32:46.4542930Z         compare_two_images(reference=os.path.join(ls49_big_data,"reference","ref_cpu_step5_000000_intimage_001.img"), test="cpu_step5_000000_intimage_001.img")
2021-12-20T00:32:46.4548374Z       File "/__w/1/modules/LS49/tests/tst_monochromatic_image.py", line 70, in compare_two_images
2021-12-20T00:32:46.4549061Z         assert N_large_deltas == 0, "%d pixels have |delta| larger than cutoff %d"%(N_large_deltas,tolerance_delta)
2021-12-20T00:32:46.4549716Z     AssertionError: 23 pixels have |delta| larger than cutoff 50
2021-12-20T00:32:46.4550223Z   Standard error:
2021-12-20T00:32:46.4550570Z     Traceback (most recent call last):
2021-12-20T00:32:46.4558091Z       File "/__w/1/modules/LS49/tests/tst_polychromatic_image.py", line 40, in <module>
2021-12-20T00:32:46.4558846Z         reference=os.path.join(ls49_big_data,"reference","step5_MPIbatch_000000.img.gz"), test="./step5poly_000000.img.gz")
2021-12-20T00:32:46.4559490Z       File "/__w/1/modules/LS49/tests/tst_monochromatic_image.py", line 70, in compare_two_images
2021-12-20T00:32:46.4560361Z         assert N_large_deltas == 0, "%d pixels have |delta| larger than cutoff %d"%(N_large_deltas,tolerance_delta)
2021-12-20T00:32:46.4561103Z     AssertionError: 6002 pixels have |delta| larger than cutoff 50
2021-12-20T00:32:46.4561552Z   Standard error:
2021-12-20T00:32:46.4565529Z     Traceback (most recent call last):
2021-12-20T00:32:46.4566134Z       File "/__w/1/modules/LS49/tests/tst_cpu_add_spots_double_precision.py", line 38, in <module>
2021-12-20T00:32:46.4566844Z         compare_two_images(reference=os.path.join(ls49_big_data,"reference","ref_cpu_step5poly_000000_intimage_001.img"), test="cpu_step5poly_000000_intimage_001.img")
2021-12-20T00:32:46.4567616Z       File "/__w/1/modules/LS49/tests/tst_monochromatic_image.py", line 70, in compare_two_images
2021-12-20T00:32:46.4568641Z         assert N_large_deltas == 0, "%d pixels have |delta| larger than cutoff %d"%(N_large_deltas,tolerance_delta)
2021-12-20T00:32:46.4569238Z     AssertionError: 47 pixels have |delta| larger than cutoff 50
2021-12-20T00:32:46.4572465Z 
2021-12-20T00:32:46.4572780Z Please verify these tests manually.
2021-12-20T00:32:46.4573094Z 
2021-12-20T00:32:46.4573469Z Summary:
2021-12-20T00:32:46.4576392Z   Tests run                    : 92
2021-12-20T00:32:46.4576872Z   Failures                     : 10
2021-12-20T00:32:46.4577379Z   Warnings (possible failures) : 0
2021-12-20T00:32:46.4580117Z   Known Failures (  0)         : 0
2021-12-20T00:32:46.4580599Z   Known Unstable (  0)         : 0
2021-12-20T00:32:46.4581095Z   Stderr output (discouraged)  : 10
2021-12-20T00:32:46.4581487Z 
2021-12-20T00:32:46.4585155Z See run_tests_parallel_zlog for full output.
2021-12-20T00:32:46.4585328Z 
2021-12-20T00:32:46.4585530Z 
2021-12-20T00:32:46.4585925Z ********************************************************************************
2021-12-20T00:32:46.4586419Z ERROR: 10 TEST FAILURES.  PLEASE FIX BEFORE COMMITTING CODE.
2021-12-20T00:32:46.4589510Z ********************************************************************************
```
</details>

One possible fix is just generating new reference images with the new solvent model. For now we will disable the tests in the LS49 and xfel_regression repos, so this issue is here to give some context.

**Comment by dwpaley** [comment]
See also https://gitlab.com/cctbx/xfel_regression/-/commit/b4d9f3a3bdc1e060e72bb832d78e691f335f43f6

**Comment by dwpaley** [comment]
This was fixed by #714, #716, nksauter/LS49#11, and nksauter/LS49#12.
